core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.component.ts

```diff
@@ -253,11 +253,39 @@ export class CustomizeInteractionModalComponent
     this.isinteractionOpen = true;
   }
 
+  buttonEnabledOnEndExploration(): boolean {
+    /*
+      For the EndExploration interaction, we check whether the last input field
+      is empty or not.
+      Since checking recommended exploration ID's validity is an
+      expensive operation, it is only checked in the interaction validations
+      and not here in real-time.
+    */
+    let inputField = document.
+      getElementsByClassName('oppia-exploration-id-input');
```

Comment:
Never use document.getElementsByClassName() and angular.element(). Find another way to do this.

I think this is a form and we already have a way for forms in modals to communicate validity "up through the layers". This needs to be done in a generic way and with no special casing for a particular type of interaction (i.e. the composite form is valid if all its subfields are valid).

You might want to talk to @ashutoshc8101 and @srijanreddy98 for advice, but first it might be worth taking a look at other interactions -- how is invalid input prevented? In general, the modal should be saveable if all input is valid and not saveable otherwise.